### PR TITLE
Add ignore empty-lines to no-eol-whitespace; closes #1273

### LINKED ISSUE #1273: Add ignore: ["empty-lines"] option to no-eol-whitespace
> Describe the feature request (new option)

The rule `no-eol-whitespace` checks for trailing whitespaces, which is very good. But it complains even about "blank" lines made of whitespaces only. There are some code styles which allow (or request) such behaviour (and some tools have settings for this): some questions about this ([1](http://stackoverflow.com/questions/15051201/indent-blank-lines-intellij-idea), [2](http://stackoverflow.com/questions/5917956/why-is-indentation-in-empty-lines-bad)); [`skipBlankLines` option for `no-trailing-spaces` rule for ESLint](http://eslint.org/docs/rules/no-trailing-spaces); [`ignoreEmptyLine` option for `disallowTrailingWhitespace` rule for JSCS](http://jscs.info/rule/disallowTrailingWhitespace); [`Keep indents on empty lines` setting for WebStorm](http://imgur.com/vVVlIBM).

Though this behaviour could be controversial, I think it's good to have a way to allow it by adding option like `ignore: "blank-lines"` to `no-eol-whitespace` rule.

> What CSS is needed to reproduce this issue?

```css
.container {
••position: relative;
••
••font-weight: bold;
}
```

`•` represents whitespaces; note the 3rd line which has "trailing" spaces at the same indent as a whole block. Now you'll have error from stylelint: `21:4  ✖  Unexpected whitespace at end of line   no-eol-whitespace`. 

> What did you expect to happen?

 With option enabled: No warnings for lines made of whitespaces only:

```json
{ "rules": {
  "no-eol-whitespace": [ true, {
    "ignore": "blank-lines"
  } ]
} }
```

### PR DESCRIPTION
**Pull request by davidtheclark**
Ref #1273.

Adds a new `isOnlyWhitespace` util that I thought may come in handy elsewhere.

### CODE REVIEW

File: src/rules/no-eol-whitespace/README.md [hunk 1]
```diff
@@ -39,3 +39,30 @@ a { color: pink; }
...
  * something else */
 ```
+
+## Optional options
+
+### `ignore: ["empty-lines"]`
+
+#### `"empty-lines"`
+
+Don't disallow end-of-line whitespace for lines that are only whitespace, "empty" lines.
```
Comment (jeddy3) [inline src/rules/no-eol-whitespace/README.md]: I think "Allow end-of-line whitespace for lines that are only whitespace, "empty" lines." is more consistent with these descriptions [else where](https://github.com/stylelint/stylelint/tree/master/src/rules/selector-no-type#descendant).

File: src/utils/isOnlyWhitespace.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import isWhitespace from "./isWhitespace"
+
```
Comment (jeddy3) [inline src/utils/isOnlyWhitespace.js]: Would it be possible to get a doc block here? Or is that seen as redundant?

**Comment by jeddy3** [comment]
I had a two documentation requests, otherwise LGTM!

**Comment by davidtheclark** [comment]
Good suggestions.

**Pull request merged**